routes/api/stream/index.js

```diff
@@ -1,33 +1,17 @@
 const express = require('express');
 
+const Comment = require('../../../models/comment');
+const User = require('../../../models/user');
+const Action = require('../../../models/action');
+
 const router = express.Router();
 
 router.get('/', (req, res, next) => {
-  console.log('Stream endpoint has been hit with asset_id ', req.query.asset_id);
-  res.json([
-    {
-      'id': 'abc',
-      'type': 'comment',
-      'body': 'Sample comment',
-      'created_at': new Date().getTime(),
-      'asset_id': 'assetTest'
-    },
-    {
-      'id': 'xyz',
-      'type': 'comment',
-      'body': 'Sample reply',
-      'created_at': new Date().getTime() - 600000,
-      'parent_id': 'abc',
-      'asset_id': 'assetTest'
-    },
-    {
-      'id': 'def',
-      'type': 'comment',
-      'body': 'Another comment',
-      'created_at': new Date().getTime() - 400000,
-      'asset_id': 'assetTest'
-    }
-  ]).catch(next);
+  const comments = Comment.findByAssetId(req.query.asset_id) || [];
+  const users = User.findByIdArray(comments.map((comment) => comment.author_id));
+  const actions = Action.findByItemIdArray(comments.map((comment) => comment.id));
```

Comment:
Ok, this sums up the beauty/horribleness of javascript. `comments`, `users` and `actions` here are all `Promise`s when this code is run, and will _eventually_ resolve to arrays at some (unknown) point in the future. So below, when we call `res.json`, you'll get something crazy in the json. So to solve this, you need to pass all the Promises to `Promise.all` like 

```
Promise.all([comments, users, actions]).then(([comments, users, actions]) => {
  // comments, users, actions will be resolved to arrays here.
}).catch(error => {
  // handle error
});
```

[Promise.all MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)